File: ProyectoAsisTrack/src/main/java/com/example/ProyectoAsisTrack/Repository/AsistenciaRepository.java
package com.example.ProyectoAsisTrack.Repository;

import com.example.ProyectoAsisTrack.Model.Asistencia;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;


@Repository
public interface AsistenciaRepository extends JpaRepository<Asistencia, Long> {
}

File: app/src/main/java/com/example/proyectoasistrack/data/Repository/AsistenciaRepository.kt
package com.example.proyectoasistrack.data.repository

import com.example.proyectoasistrack.data.dao.AsistenciaDao
import com.example.proyectoasistrack.data.model.Asistencia
import kotlinx.coroutines.flow.Flow

class AsistenciaRepository(private val dao: AsistenciaDao) {

    fun getAllAsistencias(): Flow<List<Asistencia>> = dao.getAllAsistencias()

    suspend fun insert(asistencia: Asistencia) = dao.insert(asistencia)

    suspend fun update(asistencia: Asistencia) = dao.update(asistencia)

    suspend fun delete(asistencia: Asistencia) = dao.delete(asistencia)
}
